contracts/examples/OracleCreator.sol

```diff
@@ -0,0 +1,183 @@
+pragma solidity =0.6.6;
+
+import './../libraries/TransferHelper.sol';
+import './../libraries/DXswapOracleLibrary.sol';
+import './../libraries/DXswapLibrary.sol';
+import './../libraries/SafeMath.sol';
+
+contract OracleCreator {    
+    using FixedPoint for *;
+    using SafeMath for uint256;
+
+    event OracleCreated(
+        uint256 _oracleId,
+        address _pair,
+        uint256 _windowSize,
+        uint8 _granularity,
+        uint256 _periodSize,
+        uint256 _executionBountyWei
+    );
+
+    event OracleExecuted(
+        uint256 _oracleId
+    );
+
+    struct Oracle{
+      uint256 windowSize;
+      address pair;
+      uint8 granularity;
+      uint256 periodSize;
+      uint256 executionBountyWei;
+      uint256 finishedTimestamp;
+      bool observationFinished;
+    }
+
+    struct Observation {
+      uint256 timestamp;
+      uint256 price0Cumulative;
+      uint256 price1Cumulative;
+    }
+
+    address public immutable DXswapRelayer;
+    mapping(uint256 => Oracle) public oracles;
+    uint256 public oraclesIndex;
+    mapping (uint256 => Observation[]) public oracleObservations;
+    
+    constructor(
+        address payable _DXswapRelayer
+    ) public {
+        DXswapRelayer = _DXswapRelayer;
+    }
+
+    function createOracle(
+        uint256 windowSize,
+        address pair,
+        uint8 granularity,
+        uint256 executionBountyWei
+    ) public payable returns (uint256 oracleId) {
+        uint256 periodSize;
+        require(msg.sender == DXswapRelayer, 'DXliquidityRelay: CALLER_NOT_RELAYER');
+        require(granularity > 1, 'DXliquidityRelay: GRANULARITY');
+        require(
+            (periodSize = windowSize / granularity) * granularity == windowSize,
+            'DXliquidityRelay: WINDOW_NOT_EVENLY_DIVISIBLE'
+        );
+        require(msg.value >= executionBountyWei.mul(periodSize), 'DXliquidityRelay: INVALID_BOUNTY');
+
+        oracles[oraclesIndex] = Oracle(
+            windowSize,
+            pair,
+            granularity,
+            periodSize,
+            executionBountyWei,
+            0,
+            false
+        );
+
+      oracleId = oraclesIndex;
+      emit OracleCreated(oracleId, pair, windowSize, granularity, periodSize, executionBountyWei);
+      oraclesIndex++;
+    }
+
+    function update(uint256 _oracleId) public payable {
```

Comment:
Consider renaming _oracleId to _oracleIndex, since it is the index of the oracle in the array, and not an ID, just a suggestion for better understanding of the code from reading the variable names.